library/src/utils/position/setTaskListWithAutoPosition.ts

```diff
@@ -0,0 +1,41 @@
+import type { BaseTask } from '../../types/baseTask';
+import type { ReturnTaskType } from '../../types/taskReturnType';
+import getTaskVerticalSizeStyle from './getTaskVerticalSizeStyle';
+
+export default function setTaskListWithAutoPosition<T extends BaseTask>({
+  groupedTaskList,
+  startTime,
+  endTime,
+}: setTaskListWithAutoPositionParameter<T>): ReturnTaskType<T>[][] {
+  const groupedTaskListWithAutoPosition = groupedTaskList.map((group) => {
+    const autoPositionedGroup = group.map((task, index) => {
+      const taskVerticalStyle = getTaskVerticalSizeStyle(
+        task,
+        startTime,
+        endTime,
+      );
+
+      const taskStyle = {
+        position: 'absolute',
+        left: `${(100 / group.length) * index}%`,
+        width: `${100 / group.length}%`,
+        ...taskVerticalStyle,
+      } as React.CSSProperties;
+
+      return {
+        ...task,
+        style: taskStyle,
+      };
+    });
```

Comment:
```suggestion
      const style = {
        position: 'absolute',
        left: `${(100 / group.length) * index}%`,
        width: `${100 / group.length}%`,
        ...taskVerticalStyle,
      };

      return {
        ...task,
        style,
      };
    });
```
이렇게 써도 깔끔할 것 같아요~
추가적으로 `taskStyle`에 달려있던 `as React.CSSProperties` 타입 단언이 꼭 필요할까? 라는 생각이 듭니다!
이미 상단에서 리턴 타입을 명시해주어서 추가적인 타입 단언이 필요한 이유를 잘 모르겠습니다

Reply:
```

      const style: React.CSSProperties = {
        position: 'absolute',
        left: `${(100 / group.length) * index}%`,
        width: `${100 / group.length}%`,
        ...taskVerticalStyle,
      };
```

해당 코드를 말씀해주신대로 변경하였습니다!
as 문을 사용했었던 이유는 typescript에서 해당 코드를 CSSProperties로 인지하지 않아서였습니다!

Reply:
단언 보다는 선언을 쓰는것이 타입 검증 측면에서 더 이점이 있다라는 이야기를 들었던 적이 있는 것 같아요!

제가 타입이 안찍혀서 인지하지 못하는지 확인을 못했군요..! 수정 감사합니다!